### PR TITLE
[strong] Switch to iojs2 and use more native features

### PR DESCRIPTION
**Pull request by arv**

### CODE REVIEW

File: src/codegeneration/ObjectLiteralTransformer.js [hunk 1]
```diff
@@ -311,6 +311,11 @@ export class ObjectLiteralTransformer extends TempVarTransformer {
   }
 
   transformPropertyMethodAssignment(tree) {
+    if (!this.needsAdvancedTransform &&
```
Comment (arv) [inline src/codegeneration/ObjectLiteralTransformer.js]: I plan to move the changes to this file to master.

File: src/syntax/LiteralToken.js [hunk 2]
```diff
@@ -123,6 +119,10 @@ class StringParser {
         return ch;
     }
   }
+
+  [Symbol.iterator]() {
```
Comment (arv) [inline src/syntax/LiteralToken.js]: In V8 it is always better to keep the computed property names last.

Comment (johnjbarton) [inline src/syntax/LiteralToken.js]: I guess you mean: In V8 we get better performance is the properties with computed names are added after all of the static names.

I think this is worth a comment, at least until its common knowledge or we have a lot of such cases.

**Comment by johnjbarton** [comment]
LGTM. "more native" in the commit message could be more specific in case we are trying to track down issues. 

The 'more native' bit must imply some restrictions on where the resulting generated code can run, eg it will run only on iojs2, correct?

**Pull request merged**